updatesnap/updatesnapyaml.py

```diff
@@ -63,9 +64,54 @@ def get_yaml_file(self, project_url, yaml_path):
                 data = None
         return data
 
+    def get_readme_file(self, project_url, readme_path):
+        """ Searches in a project for the 'README' file and
+            returns its contents """
+        try:
+            data = self._github.get_file(project_url, readme_path)
+        except (ValueError, ConnectionError):
+            data = None
+        return data
 
-def main():
-    """ Main code """
+    def update_readme(self, project_url, parts, readme_path):
+        """
+        Updates the <!-- Begin Included Components --> to <!-- End Included Components -->
+          sections in the README with the updated part list.
+        """
+        if readme_path is None:
+            return
+
+        readme_data = self.get_readme_file(project_url, readme_path)
+        if not readme_data:
+            print('Failed to get the README file.', file=sys.stderr)
+            sys.exit(1)
+        readme_content = readme_data.decode('utf-8')
+
+        parts_contents = "\n".join([
+            f"  - {part['name']} "
+            f"{part['updates'][0]['name'] if part['updates'] else part['version'][0]}"
+            for part in parts if part
+        ])
+        formatted_contents = f"## Included Components\n{parts_contents}"
+        updated_readme = re.sub(
```

Comment:
Maybe a check for the existence of the tags would be a good idea. That allows to ensure that they are correct (a typo would result in the data not being updated).